### ISSUE TITLE
Error: Something went wrong

### ISSUE THREAD

**Issue by MizanurRemon** [Snapchat/creative-kit-sample#53]
I receive the error message when sharing content to Snapchat.
![Screenshot_20220414_013858_com snapchat android](https://user-images.githubusercontent.com/50514672/163257307-d7f58847-3a79-4131-87a3-d22cad9a756e.jpg)

**Comment by m-mirghafari** [comment]
I got this error too. Mizanur, did you found any solution for it?

**Comment by LiveEYE074** [comment]
Nope

**Comment by xinayida** [comment]
Anyupdate abuot this issue?

**Comment by ansh** [comment]
This comment helps solve it: https://github.com/Snapchat/creative-kit-sample/issues/33#issuecomment-1010529656

**Comment by blyszcz** [comment]
I've encountered the same issue where the bundle IDs for both my production and staging environments have been added to the Snapchat developer portal for my app. Initially, everything seemed to work fine when I tested it using the production key, even without adding users as test users. However, without making any changes, I tested it again today and it suddenly stopped functioning as expected. Whenever it tries to navigate to the Snapchat app, an error message pops up saying, "Something went wrong. Please try again later!" Interestingly, the issue seems to affect regular accounts, but the test account functions correctly. Can you help us identify the cause of this problem?

The technical support in this thread seems to be weak, so I'm calling out the main contributors for assistance @davidxiasc @mwestergaardsc @srisc @julien-c

**Comment by iostyle** [comment]
Issue still continues

**Comment by Kaszmir** [comment]
Hello, @iostyle,

In our experience with the snapchat_snap-kit-react-native Android native module in [Flip](https://github.com/flip-shop/), we encountered a common issue related to the "ERROR SOMETHING WENT WRONG, PLEASE TRY AGAIN" dialog message that appears after navigating to the Snapchat app. After thorough investigation, we identified the root cause, which was related to a conflict with another library using a FileProvider definition in its Manifest file.

The conflicting provider from the external library looked like this:

```
<provider
    android:name="androidx.core.content.FileProvider"
    android:authorities="${applicationId}.provider"
    android:exported="false"
    android:grantUriPermissions="true">
    <meta-data
        android:name="android.support.FILE_PROVIDER_PATHS"
        android:resource="@xml/provider_paths" />
</provider>
```

On the other hand, Snap Kit required a distinct FileProvider definition:

```
<provider
    android:name="androidx.core.content.FileProvider"
    android:authorities="${applicationId}.fileprovider"
    android:exported="false"
    android:grantUriPermissions="true">
    <meta-data
      android:name="android.support.FILE_PROVIDER_PATHS"
      android:resource="@xml/file_paths"
/>
```

Notably, the conflict arose from different authorities values while sharing the same name for the provider. Snap Kit obtains its FileProvider using the SnapUtils class:

```
public static String getFileProviderAuthority(@NonNull Context context) {
        return context.getPackageName() + ".fileprovider";
}
```

Unfortunately, we couldn't override this function to create our own FileProvider, which would have been ideal. As a solution, we had to create a custom FileProvider for the conflicting library. Creating a custom FileProvider is a straightforward process. It involves creating an empty class that extends FileProvider and adjusting the name value in the FileProvider Manifest declaration:

```
public class CustomFileProvider extends FileProvider {
}
```
```
<provider
    android:name=".CustomFileProvider"
    android:authorities="${applicationId}.provider"
    android:exported="false"
    android:grantUriPermissions="true">
    <meta-data
        android:name="android.support.FILE_PROVIDER_PATHS"
        android:resource="@xml/provider_paths" />
</provider>

```
 
In summary, make sure to double-check client ID values during such implementations. We hope this solution proves helpful to anyone facing challenges with sharing content through Snap Kit.

**Comment by donChelsea** [comment]
Issue still exists with 3.0.0
